### PR TITLE
chore(pylint) : enable invalid_name

### LINKED ISSUE #1928: pylint config: enable "invalid-name"
We enabled pylint in the project, however several code style violations already exist. There's no point in enabling rules and then not enforcing them, therefore any rule that is not already enforced is currently disabled.

One of these rules if _the one you can see in the title is this issue_.

**How to solve**
Create a pull request with 2 commits. The commits should:

1. Enable the rule in pylintrc ([see this commit for an example](https://github.com/kantord/LibreLingo/pull/1895/commits/d4c873224d1ad11710034a955c1dbfbe617c9d19))
2. Make sure that the CI still passes after creating your pull requests. If the CI fails due to the violations of the rule you've just enabled, then fix the code style mistake in the code and create a new commit. ([see this commit for an example](https://github.com/kantord/LibreLingo/pull/1895/commits/389969a5f5e3452cf9dbe35c0848e44a552ed544))

Take a look at this MR to see how your pull request should look: https://github.com/kantord/LibreLingo/pull/1895

Another (bigger) example of a similar pull request: https://github.com/kantord/LibreLingo/pull/1893

An example of how to write your commit messages:

**Why solve like this**
I have considered keeping the rules enabled but allowing the CI to pass with a certain number of pylint rule violations. I don't like this approach though, because it results in a very long and polluted output in pylint, which would render the output unusable for practical purposes.

**Additional context**
This is a good first issue because it will likely take little time to fix, and it does not require knowledge of LibreLingo or the project structure. It also does not require you to be an expert in 

Some things you might find useful:

- How to set up the project locally: https://librelingo.app/docs/#setting-up-the-development-environment
- How to run pylint locally: `poetry run pylint $(git ls-files '*.py')`
- How to fix code style issues with `black`. This can be useful to fix indentation after you changed the code: `poetry run black apps`
- Where we store our pylint config: https://github.com/kantord/LibreLingo/blob/main/pylintrc

### PR DESCRIPTION
**Pull request by LokeshShelva**
Enables the pylint config - invalid_name
fixes #1928

### CODE REVIEW

**Comment by LokeshShelva** [comment]
So, the CI/CD is passing but when I run pylint in my local machine, I can find the errors. I will change the errors for now. Let me know what to do.
Also can we add "f" as a good name in pylint, as mostly it will be used to open files?

**Comment by kantord** [comment]
> Also can we add "f" as a good name in pylint, as mostly it will be used to open files?

I wonder if that's considered good practice, do you have some source for that (perhaps an article saying best practices for naming file pointers for python or sth like that)

**Comment by kantord** [comment]
Looks like the checks are not running at all, I wonder why. Let me check out the GitHub Actions configuration

**Comment by kantord** [comment]
Can you please try to rebase this PR to `main` @LokeshShelva ? It seems like I would need to do this to run the checks: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks however, that button is not even there, so I think that it migth be that you created this from an older version of the main branch that still did not have the github actions configuration. That's the only reason I can think of right now :thinking:

**Comment by LokeshShelva** [comment]
@kantord, I think I rebased this PR.

**Comment by kantord** [comment]
> @kantord, I think I rebased this PR.



> @kantord, I think I rebased this PR.

Hi @LokeshShelva , I figured out what the problem was. Your PR was not triggering the pipelines because I accidentally set up the pipelines to only be triggered on `push`, not on `pull_request`. That's a problem because `push` is not triggered on forks, on the other hand, pull request should be.

I [fixed it,](https://github.com/LibreLingo/LibreLingo/commit/e1923fc7736f91b4a7e6e2b5f64763f14fea8eb8) but sorry about it, but you should rebase your PR again to the most up to date version of `main` in order to see the failures on this pull request.

**Comment by LokeshShelva** [comment]
@kantord. Thank you. I will work on it .

**Comment by LokeshShelva** [comment]
@kantord, So I have added the variable name `fs` as a good name in pylint as the pytest uses that name as a fixture if I am not wrong.  We could also disabling invalid name check only in the needed blocks and remove `fs` from good names list. I think this will be better.

Also I disables invalid-name check  on [`apps/librelingo_yaml_loader/librelingo_yaml_loader/_spelling.py`](https://github.com/LibreLingo/LibreLingo/blob/03ce7c4ba58b8c32f35dd0feb4f5c58fbc66cdde/apps/librelingo_yaml_loader/librelingo_yaml_loader/_spelling.py#L3). It uses the `hunspell` modules and the name does not conform with pylint.

Do you have any suggestion for these two?

**Review by kantord — CHANGES_REQUESTED** [review]
Hi, there are some small details but otherwise looks good!

if you want, you can watch my review: https://www.youtube.com/watch?v=1-MAeFngEyM

File: apps/librelingo_json_export/tests/test_integration.py [hunk 1]
```diff
@@ -8,8 +8,8 @@
 
 
 def read_json_file(path):
-    with open(path) as fp:
-        return json.load(fp)
+    with open(path) as file_path:
+        return json.load(file_path)
```
[the remark this answers is not in the document]
Comment (LokeshShelva) [reply apps/librelingo_json_export/tests/test_integration.py:12]: I think `json_file` might be a better name here

File: pylintrc [hunk 2]
```diff
@@ -320,7 +319,8 @@ good-names=i,
            k,
            ex,
            Run,
-           _
+           _,   
+           fs
```
[the remark this answers is not in the document]
Comment (LokeshShelva) [reply pylintrc]: fs is not a good name. I removed it. Now pylint is ignored at the required blocks.

**Comment by kantord** [comment]
Hi @LokeshShelva , I could not merge this because there are some conflicts. Can you please resolve them?

File: pylintrc (hunk 2 above)
[the remark this answers is not in the document]
Comment (kantord) [reply pylintrc]: perfect, thanks!

**Comment by kantord** [comment]
closing because it's stale and there was no response. feel free to reopen if you can do the conflic resolution

**Pull request closed without merging**